src/AdminSettings.vue

```diff
@@ -0,0 +1,200 @@
+			</li>
+			<li>
+				<input id="password-policy__settings__enforce-upper-lower-case"
+					v-model="config.enforceUpperLowerCase"
+					type="checkbox"
+					class="checkbox"
+					@change="updateBoolSetting('enforceUpperLowerCase')">
+				<label for="password-policy__settings__enforce-upper-lower-case">
+					{{ t(appName, 'Enforce upper and lower case characters') }}
+				</label>
+			</li>
+			<li>
+				<input id="password-policy__settings__enforce-numeric-char"
+					v-model="config.enforceNumericCharacters"
+					type="checkbox"
+					class="checkbox"
+					@change="updateBoolSetting('enforceNumericCharacters')">
+				<label for="password-policy__settings__enforce-numeric-char">
+					{{ t(appName, 'Enforce numeric characters') }}
+				</label>
+			</li>
+			<li>
+				<input id="password-policy__settings__enforce-special-char"
+					v-model="config.enforceSpecialCharacters"
+					type="checkbox"
+					class="checkbox"
+					@change="updateBoolSetting('enforceSpecialCharacters')">
+				<label for="password-policy__settings__enforce-special-char">
+					{{ t(appName, 'Enforce special characters') }}
+				</label>
+			</li>
+			<li>
+				<input id="password-policy__settings__enforce-haveibeenpwned"
+					v-model="config.enforceHaveIBeenPwned"
+					type="checkbox"
+					class="checkbox"
+					@change="updateBoolSetting('enforceHaveIBeenPwned')">
+				<label for="password-policy__settings__enforce-haveibeenpwned">
+					{{ t(appName, 'Check password against the list of breached passwords from haveibeenpwned.com') }}
+				</label>
+				<p class="havibeenpwned-hint">
+					{{ t(appName, 'This check creates a hash of the password and sends the first 5 characters of this hash to the haveibeenpwned.com API to retrieve a list of all hashes that start with those. Then it checks on the Nextcloud instance if the password hash is in the result set.') }}
+				</p>
+			</li>
+		</ul>
+	</div>
+</template>
+
+<script>
+import { loadState } from '@nextcloud/initial-state'
+
+export default {
+	name: 'AdminSettings',
+
+	data() {
+		return {
+			appName: 'password_policy',
+			config: loadState(appName, 'config'),
+		}
+	},
+
+	methods: {
+		updateBoolSetting(setting) {
+			OCP.AppConfig.setValue(this.appName, setting, this.config[setting] ? '1' : '0')
+		},
+		updateNumberSetting(setting) {
+			OC.msg.startSaving('#password-policy__saving-msg')
+
+			// If value not only (positive) numbers
+			if (!/^\d+$/.test(this.config[setting])) {
+				let message = t(appName, 'Unknown error')
+				switch (setting) {
+				case 'minLength':
+					message = t(appName, 'Minimal length has to be a non negative number')
+					break
+				case 'historySize':
+					message = t(appName, 'History size has to be a non negative number')
+					break
+				case 'expiration':
+					message = t(appName, 'Expiration days have to be a non negative number')
+					break
+				case 'maximumLoginAttempts':
+					message = t(appName, 'Maximum login attempts have to be a non negative number')
```

Comment:
I'm not sure that would be caught by the translate bot.
I think it needs to be set with the hardcoded app name

Reply:
Ahh, sure, you're right. 🤦‍♂️